Fix the issue with the pet attribute sliders not having a default val…

zehash opened the pull request:
…ue stored in them.

https://www.pivotaltracker.com/n/projects/1108894/stories/102316738

Added default values in a new migration and set them from there. Also modified the pet model.
#102316738

zero51 committed:
Merge branch 'release/0.6.1' into develop

zero51 committed:
Add image to pet list API endpoint

https://www.pivotaltracker.com/story/show/104963166

* Also change the key from an array to a single item.

zehash committed:
Fix the issue with the pet attribute sliders not having a default value stored in them.

https://www.pivotaltracker.com/n/projects/1108894/stories/102316738

Added default values in a new migration and set them from there. Also modified the pet model.

#102316738

zero51 commented on app/models/pet.rb:
```diff
@@ -10,7 +10,7 @@ class Pet < ActiveRecord::Base
   validates_presence_of :other_pet_type, if: proc {|pet| pet.pet_type_id == 5} # when pet type is 'other'
   validates_inclusion_of :pet_type_id, :in => ReferenceData::PetType.all.map(&:id)
-  validates :energy_level, inclusion: { in: 1..5 } # Low, low medium, medium, high medium, and high
+  validates :energy_level, inclusion: { in: %w(1 2 3 4 5) } # Low, low medium, medium, high medium, and high
```
So energy level should be a string, whereas size and sex IDs are integers?

zehash replied:
We are using reference data for size and sex. If you look at how we map them using reference data, then yes, I believe they should be.

zero51 commented on db/migrate/20151004053415_set_default_value_to_pet.rb:
```diff
@@ -0,0 +1,7 @@
+class SetDefaultValueToPet < ActiveRecord::Migration
+  def change
+    change_column :pets, :energy_level, :string,  :limit => 255, :default => "3"
```
You're changing the energy level type?

zehash replied:
I didnt change datatype
I just added default values

size was only integer rest two were strings

zero51 replied:
Please look at the changes to schema.rb below, or at your local DB, or in the console - `energy_level` has changed from integer to string.

```
irb(main):004:0> Pet
=> Pet(id: integer, user_id: integer, name: string, breed: string, created_at: datetime, updated_at: datetime, dislike_loneliness: boolean, dislike_children: boolean, dislike_animals: boolean, dislike_people: boolean, flea_treated: boolean, vaccinated: boolean, house_trained: boolean, age: string, microchip_number: string, council_number: string, explain_dislikes: text, other_pet_type: string, emergency_contact_name: string, emergency_contact_phone: string, vet_name: string, vet_phone: string, medication: text, date_of_birth: date, pet_type_id: integer, sex_id: integer, size_id: integer, energy_level: integer, personalities: text)
```

zehash replied:
Oh damn - I did change that data type. Accident. I just wanted to add default values. Don't merge that. Ill making the changes and sending them in. Good eye.

zehash committed:
Labels: HEAD origin/feature/HR-102316738 feature/HR-102316738

Fix the issue with the pet attribute sliders not having a default value stored in them.

https://www.pivotaltracker.com/n/projects/1108894/stories/102316738

Fixed the energy_level datatype issue.

#102316738

zero51 committed:
Add device and OAuth details to login and signup API endpoints

https://www.pivotaltracker.com/story/show/104976840

* A user can have many devices.
* Only active devices are considered for APNs.
* On login and signup, the user's current device is sent.
* The user's OAuth details are shared similarly.

zehash committed:
Merge branch 'develop' of https://github.com/pethomestay/phs into feature/HR-102316738

Conflicts:
	db/schema.rb

zehash committed:
https://www.pivotaltracker.com/n/projects/1108894/stories/102316738

Updated the schema file - ran that migration and tests seem to pass.

102316738

zero51 commented:
I think you've made a mistake with the PR. You want to merge it into master?

zero51 commented:
Please remember to use labels to identify PR status. If you want it reviewed, then set the label to 'needs review' and the assignee to me.

zehash commented:
Yea I saw that, but the reason I want too merge into develop was because today we do the weekly push where we merge develop into master

zero51 commented:
Please be careful with your commit messages. Review the style guide for best practice (see https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message).

zero51 commented:
@zehash You will **always** want to merge into develop (unless it's a hotfix). Deployments to production are handled by myself via releases.

zehash commented:
Makes sense. What do you want me to do then. Resubmit this as a hotfix?

zero51 commented:
No, definitely not. Remember that hotfixes should only be reserved for critical bug fixes. They will be flagged as such in Pivotal. The PR needs to be corrected so that the target is develop.

zehash commented:
So I should change the ticket to feature instead of bug right?

zero51 commented:
No, it's still a bug. The Pivotal story type is correct. This PR is incorrect.
